components/interactive-markdown.js

```diff
@@ -23,18 +28,49 @@ function ClickToRenderCodePreview(props) {
   )
 }
 
+function createCodeHandler(component) {
+  return function codeHandler(options, value) {
+    return {component, code: value, ...options}
+  }
+}
+
+function highlightWithPrism(code, language) {
+  return Prism.highlight(code, Prism.languages[language])
+}
+const prismHighlighter = highlighter(highlightWithPrism)
+
 function interactiveMarkdown(markdownString) {
   const componentBlocks = []
+  const highlightHtml = prismHighlighter('html')
+  const highlightJavaScript = prismHighlighter('javascript')
+  const highlightBash = prismHighlighter('bash')
+
+  const codeBlock = createCodeHandler(StaticCodeBlock)
+  const htmlCodeBlock = highlightHtml(codeBlock)
+  const javaScriptCodeBlock = highlightJavaScript(codeBlock)
+  const bashCodeBlock = highlightBash(codeBlock)
+  const codePreview = createCodeHandler(CodePreview)
+  const clickToRenderCodePreview = createCodeHandler(ClickToRenderCodePreview)
+
   const pragmaHandlers = {
     interactive(options, value) {
-      const component = options.clickToRender ?
-        ClickToRenderCodePreview :
-        CodePreview
-      return {component, code: value, ...options}
+      const interativeCodeBlock = options.clickToRender ?
+        clickToRenderCodePreview :
+        codePreview
+      return interativeCodeBlock(options, value)
     },
     callout(options, value) {
       return {component: Callout, children: value, ...options}
     },
+    js(options, value) {
```

Comment:
Great. I'll close the PR until it is ready, so you don't a commit by commit play by play. ;-)